src/models/BibDetails.ts

```diff
@@ -0,0 +1,202 @@
+  }
+
+  buildStandardDetail(fieldMapping: FieldMapping) {
+    const bibFieldValue = this.bib[fieldMapping.field]
+    return this.buildDetail(fieldMapping.label, bibFieldValue)
+  }
+
+  buildDetail(label, value) {
+    if (!value?.length) return null
+    return { label, value }
+  }
+
+  buildInternalLinkedDetail(fieldMapping: {
+    label: string
+    field: string
+  }): LinkedBibDetail {
+    const value = this.bib[fieldMapping.field]
+    if (!value?.length) return null
+    return {
+      link: "internal",
+      label: fieldMapping.label,
+      value: value.map((v: string) => {
+        const internalUrl = `/search?filters[${fieldMapping.field}][0]=${v}`
+        return { url: internalUrl, urlLabel: v }
+      }),
+    }
+  }
+
+  get extent(): BibDetail {
+    let modifiedExtent: string[]
+    const { extent, dimensions } = this.bib
+    const removeSemiColon = (extent) => [extent[0].replace(/\s*;\s*$/, "")]
+    const extentExists = extent && extent[0]
+    const dimensionsExists = dimensions && dimensions[0]
+    if (!extentExists && !dimensionsExists) return null
+    if (!extentExists && dimensionsExists) modifiedExtent = dimensions
+    if (extentExists && !dimensionsExists) {
+      modifiedExtent = removeSemiColon(extent)
+    }
+    if (extentExists && dimensionsExists) {
+      const parts = removeSemiColon(extent)
+      parts.push(dimensions[0])
+      modifiedExtent = [parts.join("; ")]
+    }
+    return {
+      label: "Description",
+      value: modifiedExtent,
+    }
+  }
+
+  get supplementaryContent(): LinkedBibDetail {
+    if (
+      !this.bib.supplementaryContent?.length ||
+      // This is not for a known edge case, just here as a safeguard to avoid
+      // broken a tags.
+      !this.bib.supplementaryContent[0]?.url
+    ) {
+      return null
+    }
+    const label = "Supplementary Content"
+    const values = this.bib.supplementaryContent.map((sc) => {
+      return {
+        url: sc.url,
+        urlLabel: sc.label,
+      }
+    })
+    return {
+      link: "external",
+      label,
+      value: values,
+    }
+  }
+
+  get subjectHeadings() {
+    const subjectLiteralUrls = this.bib.subjectLiteral.map(
+      (subject: string) => {
+        subject = subject.replace(/\.$/, "")
+        // stackedSubjectHeadings: ["a", "a -- b", "a -- b -- c"]
+        const stackedSubjectHeadings =
+          this.constructSubjectHeadingsArray(subject)
+        const filterQueryForSubjectHeading = "/search?filters[subjectLiteral]="
+        // splitSubjectHeadings: ["a", "b", "c"]
+        const splitSubjectHeadings = subject.split(" -- ")
+        return splitSubjectHeadings.map((heading, index) => {
+          const urlWithFilterQuery = `${filterQueryForSubjectHeading}${stackedSubjectHeadings[index]}`
+          const subjectHeadingUrl = {
+            url: urlWithFilterQuery,
+            urlLabel: heading,
+          }
+          return subjectHeadingUrl
```

Comment:
Again you probably don't need to instantiate a variable, you can just return the line above.

Reply:
ah yes thats leftover from debugging